core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2ServiceModule.java

```diff
@@ -40,13 +40,19 @@ protected void setup(Binder binder)
     {
         jaxrsBinder(binder).bind(OAuth2CallbackResource.class);
 
-        configBinder(binder).bindConfig(OAuth2Config.class);
         binder.bind(OAuth2Service.class).in(Scopes.SINGLETON);
         newOptionalBinder(binder, OAuth2Client.class)
                 .setDefault()
                 .to(ScribeJavaOAuth2Client.class)
                 .in(Scopes.SINGLETON);
-        httpClientBinder(binder).bindHttpClient("oauth2-jwk", ForOAuth2.class);
+
+        OAuth2Config oauth2Config = buildConfigObject(OAuth2Config.class);
+        httpClientBinder(binder)
+                .bindHttpClient("oauth2-jwk", ForOAuth2.class)
+                .withConfigDefaults(config -> {
+                    oauth2Config.getTruststorePath().map(Path::toString).ifPresent(config::setTrustStorePath);
+                    oauth2Config.getTruststorePassword().ifPresent(config::setTrustStorePassword);
```

Comment:
we should not set this if `getTruststorePath()` is empty. That case should raise a config validation error.